.github/workflows/ci.yml

```diff
@@ -0,0 +1,83 @@
+name: Alfresco Java SDK CI
+
+on:
+  push:
+    branches:
+      - develop
+      - release/**
+      - dev-**
+  pull_request:
+    branches:
+      - develop
+      - release/**
+      - dev-**
+  workflow_dispatch:
+
+env:
+  MAVEN_PASSWORD: ${{ secrets.NEXUS_PASSWORD }}
+  MAVEN_USERNAME: ${{ secrets.NEXUS_USERNAME }}
+  GITHUB_ACTIONS_DEPLOY_TIMEOUT: 60
+  CONTENT_SERVICE_SECURITY_BASICAUTH_PASSWORD: ${{ secrets.CONTENT_SERVICE_SECURITY_BASICAUTH_PASSWORD }}
+  CONTENT_SERVICE_SECURITY_BASICAUTH_USERNAME: ${{ secrets.CONTENT_SERVICE_SECURITY_BASICAUTH_USERNAME }}
+  CONTENT_SERVICE_URL: ${{ secrets.CONTENT_SERVICE_URL }}
+  ENVIRONMENT_APA_HOST: apadev-apa.envalfresco.com
+  ENVIRONMENT_HOST: apadev.envalfresco.com
+  ENVIRONMENT_APP: simpleapp
+  AGS_VERSION: 14.9
+  ACS_VERSION: 7.1.0.1
+
+
+jobs:
+  pre-commit:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: Alfresco/alfresco-build-tools/.github/actions/pre-commit@v1.35.0
+
+  build_and_verify:
+    name: "Build and Verify"
+    runs-on: ubuntu-latest
+    needs:
+      - pre-commit
+    env:
+      MAVEN_CLI_OPTS: >
+        source:jar -B -q -e -fae -V -DinstallAtEnd=true -U -Ddoclint=none
+        -Denvironment.host=${ENVIRONMENT_HOST}
+        -Denvironment.apa.host=${ENVIRONMENT_APA_HOST}
+        -Denvironment.application.name=${ENVIRONMENT_APP}
+        -Dags.version=${AGS_VERSION}
+        -Dacs.version=${ACS_VERSION}
+    steps:
+      - uses: actions/checkout@v3
+      - uses: Alfresco/alfresco-build-tools/.github/actions/get-build-info@v1.35.0
+      - uses: Alfresco/alfresco-build-tools/.github/actions/setup-java-build@v1.35.0
+      - name: "Build"
+        timeout-minutes: ${{ fromJSON(env.GITHUB_ACTIONS_DEPLOY_TIMEOUT) }}
+        run: mvn install $MAVEN_CLI_OPTS -DskipTests=true -Dmaven.javadoc.skip=true
+      - name: "Run tests"
+        timeout-minutes: ${{ fromJSON(env.GITHUB_ACTIONS_DEPLOY_TIMEOUT) }}
+        run: mvn verify $MAVEN_CLI_OPTS -Dlogging.root.level=off -Dspring.main.banner-mode=off
+
+  publish:
+    name: "Publish"
+    runs-on: ubuntu-latest
+    if: >
+      github.event_name != 'pull_request'
+    needs:
+      - build_and_verify
+    env:
+      MAVEN_CLI_OPTS: >
```

Comment:
I didn't know how to declare them globally while also using other global variables inside of them, so that's where I left them since inside of the jobs it worked correctly

Reply:
Makes sense, but did we verify that this approach actually works? I've found very recent SO threads trying to do the same thing and noticing it didn't work as expected: https://stackoverflow.com/questions/72596618/environment-variable-in-github-actions-not-interpolated-correctly.

It would be nice to have a quick test consisting of:
- declaring an env variable at the global level (e.g.: `TEST_ENV_VAR: global-env-var`)
- declaring another env variable that references the first one, at the job-level (e.g.: `COMPOSITE_ENV_VAR: this+${TEST_ENV_VAR}`)
- echoing the actual value of the second env variable within the job: `echo $COMPOSITE_ENV_VAR`

Ideally it should output the actual value `this+global-env-var` and **not** `this+${TEST_ENV_VAR}`.

I understand that the build and tests are working, so it's possible that either the approach works OR that these parameters are read directly from the environment variables OR that some of these parameters are unnecessary. So this test would let us understand better how to proceed as it would confirm whether the composition mechanism works (we can keep all environment variables as-is) or it doesn't and therefore we're probably just passing a bunch of bogus unnecessary arguments that should be pruned instead.

Reply:
In short: it works correctly

Long version: `echo $COMPOSITE_ENV_VAR` will display `this+${TEST_ENV_VAR}`, however using this variable in a `mvn verify $COMPOSITE_ENV_VAR` will be properly passed as `this+global-env-var`, so         
```
-Denvironment.host=${ENVIRONMENT_HOST}
-Denvironment.apa.host=${ENVIRONMENT_APA_HOST}
-Denvironment.application.name=${ENVIRONMENT_APP}
-Dags.version=${AGS_VERSION}
-Dacs.version=${ACS_VERSION}
```
properly passes all the env variables declared at the beginning to `mvn XXX `$MAVEN_CLI_OPTS``, while I wasn't able to achieve it with declaring MAVEN_CLI_OPTS as an env variable. To set it as a global one, we would probably have to delete declarations for all the env variables it uses and just paste their values inside of it like this:
`  MAVEN_CLI_OPTS: "source:jar -B -q -e -fae -V -DinstallAtEnd=true -U -Ddoclint=none \
  -DaltReleaseDeploymentRepository=alfresco-public::default::https://artifacts.alfresco.com/nexus/content/repositories/releases \
  -DaltSnapshotDeploymentRepository=alfresco-public-snapshots::default::https://artifacts.alfresco.com/nexus/content/repositories/snapshots \
  -Denvironment.host=apadev.envalfresco.com \
  -Denvironment.apa.host=apadev-apa.envalfresco.com \
  -Denvironment.application.name=simpleapp \
  -Dags.version=14.9 -Dacs.version=7.1.0.1"`

Reply:
Sounds good, please restore the workflow to a clean, working state 👍